pkg/common/cns-lib/volume/manager.go

```diff
@@ -2593,7 +2641,24 @@ func (m *defaultManager) deleteSnapshotWithImprovedIdempotencyCheck(
 				if volumeOperationDetails.OperationDetails.TaskStatus == taskInvocationStatusSuccess {
 					log.Infof("Snapshot id %q on Volume %q is already deleted on CNS with opId: %q.",
 						snapshotID, volumeID, volumeOperationDetails.OperationDetails.OpID)
-					return nil
+					cnsSnapshotInfo = &CnsSnapshotInfo{
+						SnapshotID:     volumeOperationDetails.SnapshotID,
+						SourceVolumeID: volumeOperationDetails.VolumeID,
+					}
+					if isStorageQuotaM2FSSEnabled && m.clusterFlavor == cnstypes.CnsClusterFlavorWorkload {
```

Comment:
From my understanding `isStorageQuotaM2FSSEnabled` will be disabled for both vanilla and guest. I don't think you need extra cluster flavor check, I maybe wrong, can you please verify

Reply:
i think it will be good to have cluster flavour check, so that in future if we remove fss it will not affect vanilla and guest